main/features.py

```diff
@@ -1,29 +1,27 @@
 """MITxOnline feature flags"""
-import os
+import posthog
 from functools import wraps
 
 from django.conf import settings
-from posthog import Posthog
-
-if "IN_TEST_SUITE" not in os.environ:
-    posthog = Posthog(settings.POSTHOG_API_TOKEN, host=settings.POSTHOG_API_HOST)
-else:
-    posthog = None
 
 
 IGNORE_EDX_FAILURES = "IGNORE_EDX_FAILURES"
 SYNC_ON_DASHBOARD_LOAD = "SYNC_ON_DASHBOARD_LOAD"
 ENABLE_ADDL_PROFILE_FIELDS = "ENABLE_ADDL_PROFILE_FIELDS"
 ENABLE_NEW_DESIGN = "mitxonline-new-product-page"
+ENABLE_NEW_HOME_PAGE_FEATURED = "mitxonline-new-featured-carousel"
+ENABLE_NEW_HOME_PAGE_HERO = "mitxonline-new-featured-hero"
+ENABLE_NEW_HOME_PAGE_VIDEO = "mitxonline-new-home-page-video-component"
 
 
-def is_enabled(name, default=None):
+def is_enabled(name, default=None, unique_id=settings.HOSTNAME):
     """
     Returns True if the feature flag is enabled
 
     Args:
         name (str): feature flag name
         default (bool): default value if not set in settings
+        unique_id (str): person identifier passed back to posthog
```

Comment:
How should a developer decide whether to set this value to `settings.HOSTNAME` or `User.id`, or maybe something else?

Reply:
That's a good question - what I went with for the home page, considering the output in the posthog application itself, was either the user's readable ID (in this case, email) OR, in the case of an anonymous user, a uuid generated per session - so we can track if/when something has gone awry. This would also allow, in the future, to filter by user if need be (if we're passing user as email, then we know, for example, to add a rule for jenniw@mit.edu rather than hostname which can vary).

Reply:
I would prefer we change all of them to use that, but for the time being, went in this direction. If you think it'd be ok to swap them over, I can, but I wasn't sure all places would have access to request/session/user.

Reply:
Makes sense.  I would add a description of how this value is used in PostHog to the argument description you currently have.  Maybe even suggest the policy you defined for using user ID or UUID so that PostHog isn't filled with a bunch of values derived from different developer's functionality.

Reply:
You could also add a description of your policy within here: https://docs.google.com/document/d/1OQhnF1t00s0yhHWSZTWSoduHlF9aHo7YWZILF5LlX1M/edit?pli=1#heading=h.2wm6l084bpzh

Reply:
Yes - Going to do that now. Sorry, a lot of this didn't become super clear until we actually started to USE the platform and some of these uses are future use, not now. I will add a descrip in the argument description and after can help flesh out the doc more.  I have been meaning to get in there, but keep finding things that need fixing :(